app/models/services/tumblr.rb

```diff
@@ -16,12 +16,15 @@ def consumer_secret
   def post(post, url='')
     body = build_tumblr_post(post, url)
     user_info = JSON.parse(client.get("/v2/user/info").body)
-    blogs = user_info["response"]["user"]["blogs"].map { |blog| URI.parse(blog['url']) }
+    blogs = user_info["response"]["user"]["blogs"]
     tumblr_ids = {}
     blogs.each do |blog|
-      resp = client.post("/v2/blog/#{blog.host}/post", body)
-      if resp.code == "201"
-        tumblr_ids[blog.host.to_s] = JSON.parse(resp.body)["response"]["id"]
+      if blog["primary"]
+        blogurl = URI.parse(blog["url"])
+        resp = client.post("/v2/blog/#{blogurl.host}/post", body)
+        if resp.code == "201"
+          tumblr_ids[blog.host.to_s] = JSON.parse(resp.body)["response"]["id"]
+        end
       end
```

Comment:
I think we can make this a bit clearer by using `blog = user_info["response"]["user"]["blogs"].find {|blog| blog["primary"] }` and dropping the `.each` loop. Btw. Are we guaranteed to have a primary blog? Or is it for example possible that only one result is returned and then it's not marked as primary?